crates/api_crud/src/post/create.rs

```diff
@@ -86,6 +86,9 @@ pub async fn create_post(
   let community = Community::read(&mut context.pool(), data.community_id).await?;
   check_community_user_action(&local_user_view.person, &community, &mut context.pool()).await?;
 
+  // If its an NSFW community, then use that as a default
+  let nsfw = Some(data.nsfw.unwrap_or_default() || community.nsfw);
```

Comment:
This should have the same effect and looks simpler:

```suggestion
  let nsfw = data.nsfw.or(Some(community.nsfw));
```